### PR TITLE
TW-888: Add unit test for `ContactsManager`

### PR DESCRIPTION
**Pull request by nqhhdev**

### CODE REVIEW

**Comment by drminh2807** [comment]
It good for Unit Test of Contact Manger
but cant test the Interactor
what do you think if we create Intergration test
just mock the datasource, then can test Repo, Interactor, ContactManger
we don't write much test for now, so write 1 test can wrap many cases maybe better

File: test/domain/contacts/contacts_manager_test.dart [hunk 1]
```diff
@@ -0,0 +1,444 @@
...
+    );
+  });
+
+  group('ContactsManager Unit test - ENV: WEB', () {
+    late ContactsManager contactsManager;
+    late MockGetTomContactsInteractor mockGetTomContactsInteractor;
+    late MockPhonebookContactInteractor mockPhonebookContactInteractor;
+
+    test(
+      'WHEN it is not available get Phonebook contact.\n'
```
Comment (hoangdat) [inline test/domain/contacts/contacts_manager_test.dart:105]: AND tomInteractor will return GetContactsSuccess with empty list

File: lib/domain/contact_manager/contacts_manager.dart [hunk 2]
```diff
@@ -12,18 +11,23 @@ import 'package:flutter/foundation.dart';
...
 
   final ValueNotifier<Either<Failure, Success>> _contactsNotifier =
       ValueNotifier(const Right(ContactsInitial()));
 
   final ValueNotifier<Either<Failure, Success>> _phonebookContactsNotifier =
       ValueNotifier(const Right(GetPhonebookContactsInitial()));
 
+  ContactsManager({
+    required this.getTomContactsInteractor,
+    required this.phonebookContactInteractor,
```
Comment (hoangdat) [inline lib/domain/contact_manager/contacts_manager.dart:28]: why not keep it as before?

**Review by hoangdat — APPROVED** [review]

**Comment by hoangdat** [comment]
please rebase it

**Comment by nqhhdev** [comment]
> please rebase it

Done

**Pull request merged**